controllers/nginx/pkg/template/template.go

```diff
@@ -323,12 +323,12 @@ func buildProxyPass(host string, b interface{}, loc interface{}) string {
 			// path has a slash suffix, so that it can be connected with baseuri directly
 			bPath := fmt.Sprintf("%s%s", path, "$baseuri")
 			if len(location.Rewrite.BaseURLScheme) > 0 {
-				abu = fmt.Sprintf(`subs_filter '<head(.*)>' '<head$1><base href="%v://$http_host%v">' r;
-	    subs_filter '<HEAD(.*)>' '<HEAD$1><base href="%v://$http_host%v">' r;
+				abu = fmt.Sprintf(`subs_filter '<head(.*?)>' '<head$1><base href="%v://$http_host%v">' r;
```

Comment:
This does not totally solve the problem. See https://github.com/kubernetes/ingress-nginx/issues/1626#issuecomment-340550711

Reply:
(Moved from #1626 )
A proposed regular expression is:

    <(?:H|h)(?:E|e)(?:A|a)(?:D|d)(?:[^">]|"[^"]*")*>

with the replacement:

    $1<base href="%v://$http_host%v">

The expression finds a "<" followed by "head" in any case, and then skip any characters that are not a quote nor ">". In case a quote is found, it skips anything until next quote, and then repeats until the ending ">" is found and stops.

Additional optimization:

- Use option "o" to replace the first match only (can it be two o more "head" tags?)

Reply:
Thank you for the excellent expression and explains!  I have added a new commit with the new expression and option "o".

For two "head" tags, replace the first match is enough. base tag is effective, and browser (tested with Chrome) will merge the two "head" tags to one.